### DISCUSSION TITLE
Is there a good way to get redis command option types?

### DISCUSSION THREAD

**Discussion by willscripted** [NodeRedis/node_redis#2609] [General]
Would love it if command option types were exported. I find myself adding option types to the top of any file that calls redis. 

For example, a recent file header adds this:

```
// From Redis
// https://github.com/redis/node-redis/blob/redis%404.6.8/packages/client/lib/commands/XADD.ts#L5-L13C2
interface XAddOptions {
    NOMKSTREAM?: true;
    TRIM?: {
        strategy?: 'MAXLEN' | 'MINID';
        strategyModifier?: '=' | '~';
        threshold: number;
        limit?: number;
    };
}
```

Is there a better way of doing this? My go-to here is using `@types` to install the missing types, but [@types/redis](https://www.npmjs.com/package/@types/redis) has the warning "This is a stub types definition. redis provides its own type definitions, so you do not need this installed." And rightly so, they seem like they exist, they just aren't exported to the public.

Has anyone found a good way of handling this?

**Comment by leibale** [comment]
You can use the `Parameters` utility type to extract it, for example:
```typescript
type XAddOptions = Parameters<typeof client['xAdd'])[3];
```
this is not optimal, and will improve in v5 :)

**Reply by willscripted** [reply]
Ah nice, `Parameters` seems like exactly what I want.
And looking forward to v5. Ty!
